comp/logs/agent/agentimpl/agent_serverless_init.go

```diff
@@ -58,5 +60,11 @@ func (a *logAgent) SetupPipeline(
 
 // buildEndpoints builds endpoints for the logs agent
 func buildEndpoints(coreConfig pkgConfig.Reader) (*config.Endpoints, error) {
-	return config.BuildServerlessEndpoints(coreConfig, intakeTrackType, config.DefaultIntakeProtocol)
+	config, err := config.BuildServerlessEndpoints(coreConfig, intakeTrackType, config.DefaultIntakeProtocol)
+	if err != nil {
+		return nil, err
+	}
+	// in serverless mode, we never want the batch strategy to flush with a tick
+	config.BatchWait = 365 * 24 * time.Hour
+	return config, nil
```

Comment:
reimplementing this: https://github.com/DataDog/datadog-agent/pull/16748/files#diff-0911ef355640c229685ea6eb45c61fdbb2855c72ad78623656a23c854e6f6eac